src/Servers/HttpSys/src/RequestProcessing/RequestContext.cs

```diff
@@ -322,5 +323,36 @@ private void SetFatalResponse(int status)
             Response.ContentLength = 0;
             Dispose();
         }
+
+        internal unsafe void Transfer(DelegationRule wrapper)
+        {
+            if (Response.HasStarted)
+            {
+                throw new InvalidOperationException("This request cannot be transfered, the response has already started.");
+            }
+
+            Response.Transfer();
+            var source = Server.RequestQueue;
+
+            var handle = GCHandle.Alloc(wrapper.Uri, GCHandleType.Pinned);
```

Comment:
I don't think I've ever seen a URI pinned to native code... Why aren't we just pinning a string here and using fixed? That would solve some issues with the codez (like if there was an exception currently calling these APIs this object will leak).